wire-ios-data-model/Source/Model/User/Availability/Availability+WireProtos.swift

```diff
@@ -0,0 +1,53 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import WireProtos
+
+extension Availability {
+
+    public init(proto: WireProtos.Availability) {
+        switch proto.type {
+        case .none:
+            self = .none
+        case .available:
+            self = .available
+        case .away:
+            self = .away
+        case .busy:
+            self = .busy
+        }
+    }
+}
+
+extension WireProtos.Availability {
```

Comment:
**(blocking)suggestion:** Aren't these same? Would moving it to a common place and import where ever these are used help in removing this duplication? We do import WireProtos in WireDataModel. I guess we could reuse that. Am I missing something here?

Reply:
`WireProtos.Availability` is generated from Rust code and is a struct, which cannot be used in a Core Data entity. That's why there is another enum which can be used by Obj-C code.
```
@objc(ZMAvailability)
public enum Availability: Int, CaseIterable {
    case none, available, busy, away
}
```